x/wasm/keeper/keeper.go

```diff
@@ -804,7 +789,24 @@ func (k Keeper) setContractInfoExtension(ctx sdk.Context, contractAddr sdk.AccAd
 
 // handleContractResponse processes the contract response
 func (k *Keeper) handleContractResponse(ctx sdk.Context, contractAddr sdk.AccAddress, ibcPort string, res *wasmvmtypes.Response) ([]byte, error) {
-	return k.wasmVMResponseHandler.Handle(ctx, contractAddr, ibcPort, res.Submessages, res.Messages, res.Data)
+	return k.doHandleWasmVMResponse(ctx, contractAddr, ibcPort, res.Submessages, res.Messages, res.Attributes, res.Data)
```

Comment:
Why this extra function.
`k.handleContractResponse` -> `k.doHandleWasmVMResponse` ?

Can't you just combine them and add this info in the function body of `handleContractResponse`?